deps/rabbit/src/rabbit_queue_member_eval.erl

```diff
@@ -0,0 +1,184 @@
+%% This Source Code Form is subject to the terms of the Mozilla Public
+%% License, v. 2.0. If a copy of the MPL was not distributed with this
+%% file, You can obtain one at https://mozilla.org/MPL/2.0/.
+%%
+%% Copyright (c) 2007-2023 VMware, Inc. or its affiliates.  All rights reserved.
+%%
+
+-module(rabbit_queue_member_eval).
+
+-feature(maybe_expr, enable).
+
+-behaviour(gen_server).
+
+-export([on_node_up/1, on_node_down/1]).
+
+-export([start_link/0]).
+
+%% gen_server callbacks
+-export([init/1, handle_call/3, handle_cast/2, handle_info/2, terminate/2,
+         code_change/3]).
+
+-define(SERVER, ?MODULE).
+-define(DEFAULT_INTERVAL, 60000*60).
+-define(SHORT_INTERVAL, 10000).
+
+-define(EVAL_MSG, member_eval).
+
+-record(state, {timer_ref, default_interval, short_interval, target_group_size}).
+
+%%----------------------------------------------------------------------------
+%% Start
+%%----------------------------------------------------------------------------
+
+-spec start_link() -> rabbit_types:ok_pid_or_error().
+
```

Comment:
Nitpic: Don't need a blank line here.

Reply:
type specs provide such important information to the function that they should be as close as possible.

(this is my opinion ofc :P )